backend/src/LibExecution/Interpreter.fs

```diff
@@ -240,6 +241,42 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
             return err id $"Missing key `{key}` in {typeStr}"
         | _ -> return result
 
+    | ERecordUpdate(id, baseRecord, updates) ->
+      let! baseRecord = eval state st baseRecord
+      let types = ExecutionState.availableTypes state
+      match baseRecord with
+      | DRecord(_, fields) when List.length updates > Map.count fields ->
+        return err id "The updates list is longer than the record filedls list"
+      | DRecord(typeName, fields) ->
+        let typ = Types.find typeName types
+        let typeStr = FQTypeName.toString typeName
+        return!
+          Ply.List.foldSequentially
+            (fun r (k, v) ->
+              uply {
+                let! v = eval state st v
+                match r with
+                | DRecord(_, m) when not (Map.containsKey k m) ->
+                  return err id $"Unexpected field `{k}` in {typeStr}"
+                | DRecord(typeName, m) ->
+                  match recordMaybe typeName with
+                  | Some(expected1, expectedRest) ->
+                    let field =
+                      (expected1 :: expectedRest)
+                      |> List.map (fun f -> f.name, f.typ)
+                      |> Map
+                      |> Map.find k
+                    match TypeChecker.unify [ k ] types field v with
+                    | Ok() -> return DRecord(typeName, Map.add k v m)
+                    | Error e -> return err id (TypeChecker.Error.toString e)
+                  | None ->
+                    return
+                      err id $"Expected a record but {typeStr} is something else"
+                | _ -> return err id "Expected a record"
```

Comment:
If a previous step returns a DError, this will replace it with a new "Expected a record" DError. We should instead handle DError and DIncomplete separately to ensure they are passed through.

Reply:
@pbiggar I'm not sure that I understand how to "handle `DError` and `DIncomplete` separately" could you please clarify what it means?

Reply:
Similar to this: https://github.com/darklang/dark/blob/ebaaedf23f63b980cc22fb9ee604f19730869ef2/backend/src/LibExecution/RuntimeTypes.fs#L663

Reply:
Basically, if we've already created a DError as part of some error handling, we should continue to propagate it through rahter than creating a new error with a different message on line 275.

Reply:
> Basically, if we've already created a DError as part of some error handling, we should continue to propagate it through rahter than creating a new error with a different message on line 275.

is it correct now?